pkg/controllers/vrep/replication_status_reconciler.go

```diff
@@ -0,0 +1,279 @@
+	TargetInfo   *ReplicationInfo
+}
+
+func MakeReplicationStatusReconciler(cli client.Client, r *VerticaReplicatorReconciler, vrep *v1beta1.VerticaReplicator,
+	log logr.Logger) controllers.ReconcileActor {
+	return &ReplicationStatusReconciler{
+		Client:     cli,
+		VRec:       r,
+		Vrep:       vrep,
+		Log:        log.WithName("ReplicationStatusReconciler"),
+		TargetInfo: &ReplicationInfo{},
+	}
+}
+
+func (r *ReplicationStatusReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
+	// no-op if replication is not being done asynchronously
+	if r.Vrep.Spec.Mode != replicationModeAsync {
+		r.Log.Info("Stopped reconciling status: not async")
+		return ctrl.Result{}, nil
+	}
+
+	// no-op if ReplicationComplete is present (either true or false)
+	isPresent := r.Vrep.IsStatusConditionPresent(v1beta1.ReplicationComplete)
+	if isPresent {
+		r.Log.Info("Stopped reconciling status: complete")
+		return ctrl.Result{}, nil
+	}
+
+	// no-op if ReplicationReady is false
+	isFalse := r.Vrep.IsStatusConditionFalse(v1beta1.ReplicationReady)
+	if isFalse {
+		r.Log.Info("Stopped reconciling status: not ready")
+		return ctrl.Result{}, nil
+	}
+
+	// no-op if there is no transaction ID we can use to query replication status
+	if r.Vrep.Status.TransactionID == 0 {
+		r.Log.Info("Stopped reconciling status: transaction ID 0")
+		return ctrl.Result{}, nil
+	}
+
+	// fetch target VerticaDB
+	if res, fetchErr := r.fetchTargetVdb(ctx); verrors.IsReconcileAborted(res, fetchErr) {
+		return res, fetchErr
+	}
+
+	// determine usernames and passwords
+	err := r.determineUsernameAndPassword(ctx)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// collect pod facts for target sandbox (or main cluster)
+	err = r.collectPodFacts(ctx)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// choose the source host and target host
+	// (first host where db is up in the specified cluster)
+	err = r.determineTargetHosts()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// build all the opts
+	opts := r.buildOpts()
+
+	// setup dispatcher for vclusterops API
+	err = r.makeDispatcher()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	err = r.runReplicationStatus(ctx, r.dispatcher, opts)
+
+	return ctrl.Result{}, err
+}
+
+// fetch the source and target VerticaDBs
+func (r *ReplicationStatusReconciler) fetchTargetVdb(ctx context.Context) (res ctrl.Result, err error) {
+	vdb := &vapi.VerticaDB{}
+	nm := names.GenNamespacedName(r.Vrep, r.Vrep.Spec.Target.VerticaDB)
+	res, err = vk8s.FetchVDB(ctx, r.VRec, r.Vrep, nm, vdb)
+	if verrors.IsReconcileAborted(res, err) {
+		return res, err
+	}
+
+	r.TargetInfo.Vdb = vdb
+	return
+}
+
+// makeDispatcher will create a Dispatcher object based on the feature flags set.
+func (r *ReplicationStatusReconciler) makeDispatcher() error {
+	if vmeta.UseVClusterOps(r.TargetInfo.Vdb.Annotations) {
+		r.dispatcher = vadmin.MakeVClusterOpsWithTarget(r.Log, r.TargetInfo.Vdb, r.TargetInfo.Vdb,
```

Comment:
Why we pass in r.TargetInfo.Vdb twicie. Can we remove one vdb from the function parameter list?